malfeasance/handler.go

```diff
@@ -235,17 +235,17 @@ func (h *Handler) validateMultipleATXs(logger log.Log, proof *types.MalfeasanceP
 			firstNid = msg.SmesherID
 			firstMsg = msg
 		} else if msg.SmesherID == firstNid {
-			if msg.InnerMsg.Target == firstMsg.InnerMsg.Target &&
-				msg.InnerMsg.MsgHash != firstMsg.InnerMsg.MsgHash {
+			if msg.InnerMsg.PublishEpoch == firstMsg.InnerMsg.PublishEpoch &&
+				msg.InnerMsg.ID != firstMsg.InnerMsg.ID {
 				return msg.SmesherID, nil
 			}
 		}
 	}
 	logger.With().Warning("received invalid atx malfeasance proof",
-		log.Stringer("smesher", ap.Messages[0].SmesherID),
-		log.Stringer("smesher", ap.Messages[1].SmesherID),
-		log.Object("first", &ap.Messages[0].InnerMsg),
-		log.Object("second", &ap.Messages[1].InnerMsg),
+		log.Stringer("first_smesher", ap.Messages[0].SmesherID),
+		log.Object("first_proof", &ap.Messages[0].InnerMsg),
+		log.Stringer("second_smesher", ap.Messages[1].SmesherID),
+		log.Object("second_proof", &ap.Messages[1].InnerMsg),
```

Comment:
why does the order need to change? and why only change atx and not other malfeasance proof?
there is no practical reason to change this. please revert.

Reply:
I had a different in-between version here. The original code logs 2 fields called "smesher" which I don't think is valid json. Ether the second value will just not be logged or the output will look like:

```json
{
    "smesher": "3fd5a",
    "smesher": "3fd5a",
    "first": { ... },
    "second": { ... },
}
```

which a lot of parsers will struggle with.

Reply:
great. can you change it also for other types, ballot and hare please